tests/components/august/test_init.py

```diff
@@ -1,136 +1,146 @@
     except HomeAssistantError as err:
         last_err = err
     assert (
         str(last_err)
-        == "mocklockid1 Name: This should bubble up as its user consumable"
+        == "A6697750D607098BAE8D6BAA11EF8063 Name: This should bubble up as its user consumable"
     )
 
 
-def test_lock_throws_august_api_http_error():
-    """Test lock."""
-    data = MockAugustComponentData(api=MockAugustApiFailing())
-    lock = _mock_august_lock()
-    data.set_mocked_locks([lock])
+async def test_lock_throws_august_api_http_error(hass):
+    """Test lock throws correct error on http error."""
+    mocked_lock_detail = await _mock_operative_august_lock_detail(hass)
+
+    def _lock_return_activities_side_effect(access_token, device_id):
+        raise AugustApiHTTPError("This should bubble up as its user consumable")
+
+    await _create_august_with_devices(
+        hass,
+        [mocked_lock_detail],
+        api_call_side_effects={
+            "lock_return_activities": _lock_return_activities_side_effect
+        },
+    )
     last_err = None
+    data = {ATTR_ENTITY_ID: "lock.a6697750d607098bae8d6baa11ef8063_name"}
     try:
-        data.unlock("mocklockid1")
+        await hass.services.async_call(LOCK_DOMAIN, SERVICE_LOCK, data, blocking=True)
     except HomeAssistantError as err:
         last_err = err
     assert (
         str(last_err)
-        == "mocklockid1 Name: This should bubble up as its user consumable"
+        == "A6697750D607098BAE8D6BAA11EF8063 Name: This should bubble up as its user consumable"
     )
 
 
-def test_inoperative_locks_are_filtered_out():
+async def test_inoperative_locks_are_filtered_out(hass):
     """Ensure inoperative locks do not get setup."""
-    august_operative_lock = _mock_operative_august_lock_detail("oplockid1")
-    data = _create_august_data_with_lock_details(
-        [august_operative_lock, _mock_inoperative_august_lock_detail("inoplockid1")]
+    august_operative_lock = await _mock_operative_august_lock_detail(hass)
+    august_inoperative_lock = await _mock_inoperative_august_lock_detail(hass)
+    await _create_august_with_devices(
+        hass, [august_operative_lock, august_inoperative_lock]
     )
 
-    assert len(data.locks) == 1
-    assert data.locks[0].device_id == "oplockid1"
-
-
-def test_lock_has_doorsense():
-    """Check to see if a lock has doorsense."""
-    data = _create_august_data_with_lock_details(
-        [
-            _mock_doorsense_enabled_august_lock_detail("doorsenselock1"),
-            _mock_doorsense_missing_august_lock_detail("nodoorsenselock1"),
-            RequestException("mocked request error"),
-            RequestException("mocked request error"),
-        ]
+    lock_abc_name = hass.states.get("lock.abc_name")
+    assert lock_abc_name is None
+    lock_a6697750d607098bae8d6baa11ef8063_name = hass.states.get(
+        "lock.a6697750d607098bae8d6baa11ef8063_name"
     )
+    assert lock_a6697750d607098bae8d6baa11ef8063_name.state == STATE_LOCKED
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5068548)